modin/engines/dask/pandas_on_dask/frame/axis_partition.py

```diff
@@ -68,8 +70,6 @@ def deploy_func_between_two_axis_partitions(
             *partitions,
             pure=False,
         )
-        if num_splits == 1:
```

Comment:
Why is this removed? We have the code that handles returned not list values (see [here](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/axis_partition.py#L94)). Also, why is not `lengths` considered here?

Reply:
Now there is no case when the type is different from the `list`.(Simplification)

Reply:
The tests did not fall - so I did not pay attention.

I will see how to properly process `lengths`.

Reply:
I think there is no reason to add length handling to `deploy_func_between_two_axis_partitions` because two frames will be combined into one by some binary operation.

Reply:
I see the point. There is no reason.